src/class-parsely.php

```diff
@@ -653,4 +670,93 @@ private function are_credentials_managed(): bool {
 
 		return true;
 	}
+
+	/**
+	 * Sets the values of managed options.
+	 *
+	 * @since 3.9.0
+	 * @access private
+	 */
+	private function set_managed_options(): void {
+		$managed_options = apply_filters( 'wp_parsely_managed_options', false );
+
+		if ( ! is_array( $managed_options ) ) {
+			return;
+		}
```

Comment:
This won't help with empty arrays, since it checks whether the user actually passed a non-array. Will add an early return after the `unsets` though if the array is empty.

Reply:
Done in https://github.com/Parsely/wp-parsely/pull/1724/commits/8fefc99de4d93c9443ae79943cc6ee541d18ed48